.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TimeFunctions.cs

```diff
@@ -76,7 +76,7 @@ public DateTimeResolutionResult PackTimeResult(DateTimeExtra<TimeType> extra, Ti
                 noDesc = false;
             }
 
-            int hour = timeResult.Hour > 0 ? timeResult.Hour : 0,
+            int hour = timeResult.Hour > 0 ? timeResult.Hour % Constants.DayHourCount : 0,

```

Comment:
What kind of case if hitting this?

Reply:
here we are normalizing the hour so that 24 -> 00